File: WebTrafficViewerJava/app/src/main/java/com/example/webtrafficviewerjava/util/JsBridge.java
package com.example.webtrafficviewerjava.util;

import android.webkit.JavascriptInterface;

public class JsBridge {

    public interface OnJsonCapturedListener {
        void onJsonCaptured(String json);
    }

    private final OnJsonCapturedListener listener;

    public JsBridge(OnJsonCapturedListener listener) {
        this.listener = listener;
    }

    @JavascriptInterface
    public void onRequestCaptured(String json) {
        if (listener != null) {
            listener.onJsonCaptured(json);
        }
    }
}

File: WebTrafficViewerJetpack/app/src/main/java/com/example/webtrafficviewerjetpack/util/JsBridge.kt
package com.example.webtrafficviewerjetpack.util

import android.webkit.JavascriptInterface

class JsBridge(
    private val onJsonCaptured: (String) -> Unit
) {
    @JavascriptInterface
    fun onRequestCaptured(json: String) {
        onJsonCaptured(json)
    }
}